qontrol/cost.py

```diff
@@ -401,7 +401,7 @@ def __call__(
         H: TimeQArray,  # noqa ARG002
         parameters: dict | Array,  # noqa ARG002
     ) -> tuple[tuple[Array, Array]]:
-        dim = jnp.prod(*result.final_propagator.dims)
+        dim = jnp.prod(jnp.array(result.final_propagator.dims))
```

Comment:
Would the simpler
```suggestion
        dim = jnp.prod(result.final_propagator.dims)
```
work?

Reply:
No, `result.final_propagator.dims` is a tuple. However, jax require only `jnp.ndarray` and do not accept implict convertion